src/lib/classes/notification.js

```diff
@@ -0,0 +1,512 @@
+
+        /**
+         * Check if settings reference was created it if not we will try one time more.
+         */
+        if (!settingsResult.ok) await this.settingsTurtle(appSettingPat, inboxPath);
+
+        /**
+         * Create a default ACL for inbox container
+         */
+        const resultAcl = await solid.fetch(`${inboxPath}.acl`, {
+          method: 'PUT',
+          body: result,
+          headers: {
+            'Content-Type': 'text/turtle'
+          }
+        });
+
+        if (!resultAcl.ok)
+          throw new SolidError(
+            result.message || 'An error when tried to assign permissions',
+            'Error',
+            resultAcl.status
+          );
+      });
+
+      return solidResponse(200, 'Inbox was created');
+    } catch (error) {
+      throw new SolidError(error.message, 'Inbox Error', error.code || error.status);
+    }
+  };
+
+  /**
+   * Fetch shape object that will be used to build notifications and also
+   * to know how we need to render it.
+   * @param file
+   * @returns {Promise<void>}
+   */
+  fetchNotificationShape = async (file, name) => {
+    try {
+      /**
+       * if shape comes like object will return the object instead of making a fetch request
+       */
+      if (typeof file === 'object') {
+        this.schema = {
+          ...this.schema,
+          [name]: file
+        };
+        return;
+      }
+      const result = await solid.fetch(file);
+      const schema = await result.json();
+
+      this.schema = {
+        ...this.schema,
+        [name]: schema
+      };
+    } catch (error) {
+      throw new SolidError(error.message, 'Fetch Shape', error.status);
+    }
+  };
+
+  buildShapeObject = shape => {
+    return {
+      name: (shape && shape.name) || this.shapeName,
+      shape: (shape && shape.path) || this.shape
+    };
+  };
+
+  /**
+   * create and send a notification to user's inbox
+   * @param inboxRoot
+   * @param title
+   * @param content
+   * @param options
+   * @returns {Promise<*>}
+   */
+
+  create = async (content = {}, to, options = {}) => {
+    try {
+      const currentShape = this.buildShapeObject(options && options.shape);
+      const { name, shape: defaultShape } = currentShape;
+      const termFactory = N3.DataFactory;
+      const { namedNode, literal } = termFactory;
+
+      if (!this.schema || (this.schema && !this.schema[name])) {
+        /**
+         * Fetch remote shape that will be use to render and build notifications
+         */
+        await this.fetchNotificationShape(defaultShape, name);
+      }
+      if (!this.schema[name]['@context']) {
+        throw new SolidError('Schema does not have context', 'Notification', 500);
+      }
+
+      const { '@context': context, shape } = this.schema[name];
+
+      /**
+       * N3 is an implementation of the RDF.js low-level specification that lets you handle RDF in JavaScript easily.
+       * Reference: https://www.npmjs.com/package/n3
```

Comment:
Great idea to link to n3! Citing your sources!

Reply:
Thank you!!